plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -34,15 +33,73 @@ import (
 
 type FailedPredicateMap map[string]sets.String
 
-type FitError struct {
+type FitResult struct {
 	Pod              *api.Pod
+	Err              chan error
 	FailedPredicates FailedPredicateMap
 }
 
+type PriorityResult struct {
+	Err             chan error
+	PrioritizeNodes schedulerapi.HostPriorityList
+}
+
+type ScheduleCache struct {
```

Comment:
I feel we do not need a cache. We just need a one direction channel that link a output to an input. Let's remove this and use a plain chan instead?